### PR TITLE
support for clustering by hostname in kubernetes 

### PR DESCRIPTION
**Pull request by pavel987**
Copying my pull request from https://github.com/aweber/rabbitmq-autocluster/pull/118

### CODE REVIEW

File: include/autocluster.hrl [hunk 1]
```diff
@@ -56,6 +56,8 @@
          {config, k8s_namespace_path,    "K8S_NAMESPACE_PATH",     "/var/run/secrets/kubernetes.io/serviceaccount/namespace",
                                                                                  string,  false},
          {config, k8s_service_name,      "K8S_SERVICE_NAME",       "rabbitmq",   string,  false},
+         {config, k8s_address_type,      "K8S_ADDRESS_TYPE",       "ip",         string,  false},
```
Comment (michaelklishin) [inline include/autocluster.hrl:59]: If the options here are `ip` and `hostname`, please leave a comment mentioning that.

Comment (michaelklishin) [reply include/autocluster.hrl:59]: Actually, the rest looks good so I will add one myself.

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
@pavel987 thank you!

**Comment by luxflux** [comment]
Thank you for your work, @pavel987 and @michaelklishin. What do you think could we have this also for RabbitMQ v3.6.x?

**Comment by michaelklishin** [comment]
@Gsantomaggio @dcorbacho does this seem like something you'd agree to backport?

**Comment by luxflux** [comment]
IMHO this is crucial to get a cluster on k8s with persistence. If using IPs, the cluster will loose it's database (incl. users, ...) sooner or later because the IPs change if a pod gets stopped/started.
But it could as well be that I did something wrong when configuring it...

I got the code working on the `stable` branch and it seems to work. Should I provide a PR...?

**Comment by Gsantomaggio** [comment]
Yes!

**Comment by michaelklishin** [comment]
@luxflux please do.

**Comment by luxflux** [comment]
PR #23 ports the changes of @pavel987 to `stable`.

**Pull request merged**